File: spring-data-jpa/src/main/java/io/beaniejoy/springdatajpa/controller/TestController.java
package io.beaniejoy.springdatajpa.controller;

import io.beaniejoy.springdatajpa.service.tx.ParentService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.io.IOException;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/test")
public class TestController {
    private final ParentService testService;

    @PostMapping
    public String transactionTest() throws IOException {
        testService.justCallChildService();
        return "ok";
    }
}

File: spring-framework-repo/src/main/kotlin/io/beaniejoy/springframeworkrepo/controller/TestController.kt
package io.beaniejoy.springframeworkrepo.controller

import com.fasterxml.jackson.annotation.JsonIgnore
import mu.KLogging
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/test")
class TestController {
    companion object: KLogging()

    @GetMapping
    fun getTest(@RequestBody temp: Temp): Temp {
        logger.info { "temp $temp" }
        return Temp("beanie", "joy")
    }
}

data class Temp(
    val temp1: String,
    val temp2: String,
) {
    @JsonIgnore
    fun getTempGetter(): String {
        return "hello"
    }
}